### PR TITLE
Add EIP: Identity aggregated NFT

### PR DESCRIPTION
**Pull request by xuxinlai2002**
…ity aggregation in NFTs (SBTs included)

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-XXXX.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+---
+title: Identity aggregated NFT, self soveriegn and composible identity aggregation in NFTs (SBTs included)
```
Comment (abcoathup) [inline EIPS/eip-XXXX.md:2]:
```suggestion
eip: 7231
title: Identity aggregated NFT, self soveriegn and composible identity aggregation in NFTs (SBTs included)
```
Issuing EIP number as PR number.  Please also update the filename

Comment (xuxinlai2002) [reply EIPS/eip-XXXX.md:2]: Ok, got it, thanks a lot.

File: EIPS/eip-7231.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
+---
+eip: 7231
+title: Identity aggregated NFT
+description: The aggregation of web2 & web3 identities to NFTs, authorized by individuals, gives attributes of ownerships, relationships, experiences.
+author: Chloe Gu <chloe@carv.io>, Navid X. (@navid), Victor Yu <victor@carv.io>, Archer H.
```
Comment (SamWilsn) [inline EIPS/eip-7231.md:5]: You'll want to add your GitHub username (`xuxinlai2002`) here or else you won't be able to edit your own EIP.

File: EIPS/eip-7231.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-06-25
+requires: 165, 721, 1271, 6066
+---
+
+## Abstract
+
+This standard extends [ERC-721](./eip-721.md) by binding individuals' Web2 and Web3 identities to non-fungible tokens (SBTs included). By binding multiple identities, aggregated and composible identity infomation can be verified, resulting in more beneficial onchain scenarios for individuals, such as self-authentication, social overlapping, commercial value generation from user targetting, etc.
```
Comment (SamWilsn) [inline EIPS/eip-7231.md:16]: It'd be nice to include a short _technical_ summary of the proposal in the abstract. You should answer the question of: how does this proposal attach identities to NFTs?

File: EIPS/eip-7231.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+     * MUST NOT modify the state
+     * MUST allow external calls
+     */
+    function verifyIdentitiesBinding(
+        uint256 id,address nftOwnerAddress,string[] memory userIDs,bytes32 identitiesRoot, bytes calldata signature
+    ) external returns (bool);
+}
+```
+
+This is the “ERCXXX Metadata JSON Schema” referenced above.
```
Comment (SamWilsn) [inline EIPS/eip-7231.md:87]: I don't see where this is mentioned above?

File: assets/eip-7231/hardhat.config.ts
Comment (SamWilsn) [inline assets/eip-7231/hardhat.config.ts:1]: Generally, we recommend against including build system files.

File: EIPS/eip-7231.md (hunk 2 above)
Comment (xuxinlai2002) [reply EIPS/eip-7231.md:5]: modify @navid -> @xuxinlai2002

File: EIPS/eip-7231.md (hunk 3 above)
Comment (xuxinlai2002) [reply EIPS/eip-7231.md:16]: add the technical summary  as follow：
By adding a custom schema in the metadata, and updating and verifying the schema hash in the contract, the binding of NFT and identity information is completed.

File: EIPS/eip-7231.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+## Motivation
+
+One of the most interesting aspects of Web3 is the ability to bring an individuals own identity to different applications. Even more powerful is the fact that individuals truly own their accounts without relying on centralized gatekeepers, disclosing to different apps components necessary for authentication and approved by individuals. 
+Exisiting solutions such as ENS, although open, decentralized, and make it more convenient for Ethereum-based applications, lack of data standardization and athentication of itentity due to inherent anominity. Other solutions such as SBTs rely on centralized attestors, can not prevent data tampering, nor inscribe data into the ledger itself in a privacy enabling way. Previous work such as [ERC-6066](./eip-6066.md) does not bind nor verify user identity data, thus unable to meet the user identity aggregation, user Kyc verification and other application scenarios related to user data sovereignty
```
[the remark this answers is not in the document]
Comment (xuxinlai2002) [reply EIPS/eip-7231.md:21]: Thanks，has been corrected

File: EIPS/eip-7231.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+The proposed pushes the boundaries of solving identity problems with Identity Aggregated NFT, i.e., the individual-authenticated aggregation of web2 and web3 identities to NFTs (SBTs included). 
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Every compliant contract must implement the Interface
+
+```typescript
```
[the remark this answers is not in the document]
Comment (xuxinlai2002) [reply EIPS/eip-7231.md:30]: Thanks，has been corrected

File: EIPS/eip-7231.md (hunk 4 above)
Comment (xuxinlai2002) [reply EIPS/eip-7231.md:87]: Thanks，has been corrected

File: assets/eip-7231/package.json
[the remark this answers is not in the document]
Comment (xuxinlai2002) [reply assets/eip-7231/package.json:1]: Thanks，has been corrected

File: assets/eip-7231/hardhat.config.ts
Comment (xuxinlai2002) [reply assets/eip-7231/hardhat.config.ts:1]: Thanks，has been corrected

File: EIPS/eip-7231.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
+---
+eip: 7231
+title: Identity aggregated NFT
+description: The aggregation of web2 & web3 identities to NFTs, authorized by individuals, gives attributes of ownerships, relationships, experiences.
+author: Chloe Gu <chloe@carv.io>, Navid X. (@xuxinlai2002), Victor Yu <victor@carv.io>, Archer H.
+discussions-to: https://ethereum-magicians.org/t/eip-7231-namespaced-storage-layout/14796
```
Comment (mryalamanchi) [inline EIPS/eip-7231.md:6]: Flagging this, the discussions-to link points to discussion on eip-7201 and not eip-7231.

@xuxinlai2002 might want to check this once.

File: EIPS/eip-7231.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+## Motivation
+
+One of the most interesting aspects of Web3 is the ability to bring an individuals own identity to different applications. Even more powerful is the fact that individuals truly own their accounts without relying on centralized gatekeepers, disclosing to different apps components necessary for authentication and approved by individuals. 
+Exisiting solutions such as ENS, although open, decentralized, and make it more convenient for Ethereum-based applications, lack of data standardization and authentication of identity due to inherent anominity. Other solutions such as SBTs rely on centralized attestors, can not prevent data tampering, nor inscribe data into the ledger itself in a privacy enabling way.
```
Comment (Benjimmutable) [inline EIPS/eip-7231.md:21]:
```suggestion
Exisiting solutions such as ENS, although open, decentralized, and more convenient for Ethereum-based applications, suffer from a lack of data standardization and authentication of identity due to inherent anominity. Other solutions such as SBTs rely on centralized attestors, can not prevent data tampering, and do not inscribe data into the ledger itself in a privacy enabling way. 
```

Comment (xuxinlai2002) [reply EIPS/eip-7231.md]: Thank you, it has been revised.

File: EIPS/eip-7231.md (hunk 7 above)
Comment (xuxinlai2002) [reply EIPS/eip-7231.md]: Thanks ,it has been revised.
https://ethereum-magicians.org/t/erc7231-identity-aggregated-nft/15062

File: EIPS/eip-7231.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-06-25
+requires: 165, 721, 1271
+---
+
+## Abstract
+
+This standard extends [ERC-721](./eip-721.md) by binding individuals' Web2 and Web3 identities to non-fungible tokens (SBTs included). By binding multiple identities, aggregated and composible identity infomation can be verified, resulting in more beneficial onchain scenarios for individuals, such as self-authentication, social overlapping, commercial value generation from user targetting, etc. By adding a custom schema in the metadata, and updating and verifying the schema hash in the contract, the binding of NFT and identity information is completed.
```
Comment (SamWilsn) [inline EIPS/eip-7231.md:16]: We generally prefer if abbreviations are expanded the first time they are used.

```suggestion
This standard extends [ERC-721](./eip-721.md) by binding individuals' Web2 and Web3 identities to non-fungible tokens (NFTs) and soulbound tokens (SBTs). By binding multiple identities, aggregated and composible identity infomation can be verified, resulting in more beneficial onchain scenarios for individuals, such as self-authentication, social overlapping, commercial value generation from user targetting, etc. By adding a custom schema in the metadata, and updating and verifying the schema hash in the contract, the binding of NFT and identity information is completed.
```

**Review by SamWilsn — APPROVED** [review]

**Comment by navid** [comment]
please note that the wrong Navid is being tagged in these
I am Navid Navab with the email address ***@***.***
and i am not a contributor to this particular git project.

**Pull request merged**